RecoEgamma/EgammaTools/src/SCEnergyCorrectorSemiParm.cc

```diff
@@ -0,0 +1,316 @@
+    clusterDPhiToSeed.resize(std::max(3, numberOfClusters), 0);
+    float clusterMaxDR     = 999.;
+    float clusterMaxDRDPhi = 999.;
+    float clusterMaxDRDEta = 999.;
+    float clusterMaxDRRawEnergy = 0.;
+    
+    size_t iclus = 0;
+    float maxDR = 0;
+    edm::Ptr<reco::CaloCluster> pclus;
+    const edm::Ptr<reco::CaloCluster>& theseed = sc.seed();
+    // loop over all clusters that aren't the seed  
+    auto clusend = sc.clustersEnd();
+    for( auto clus = sc.clustersBegin(); clus != clusend; ++clus ) {
+      pclus = *clus;
+      
+      if(theseed == pclus ) 
+	continue;
+      clusterRawEnergy[iclus]  = pclus->energy();
+      clusterDPhiToSeed[iclus] = reco::deltaPhi(pclus->phi(),theseed->phi());
+      clusterDEtaToSeed[iclus] = pclus->eta() - theseed->eta();
+      
+      // find cluster with max dR
+      const auto the_dr = reco::deltaR(*pclus, *theseed);
+      if(the_dr > maxDR) {
+	maxDR = the_dr;
+	clusterMaxDR = maxDR;
+	clusterMaxDRDPhi = clusterDPhiToSeed[iclus];
+	clusterMaxDRDEta = clusterDEtaToSeed[iclus];
+	clusterMaxDRRawEnergy = clusterRawEnergy[iclus];
+      }      
+      ++iclus;
+    }  
+    
+    // SET INPUTS
+    eval[0]  = vertices_->size(); //nVtx
+    eval[1]  = raw_energy; //scRawEnergy
+    eval[2]  = sc.etaWidth(); //scEtaWidth
+    eval[3]  = sc.phiWidth(); //scPhiWidth
+    eval[4]  = EcalClusterTools::e3x3(seedCluster,recHits,topo)/raw_energy; //scSeedR9
+    eval[5]  = seedCluster.energy()/raw_energy; //scSeedRawEnergy/scRawEnergy
+    eval[6]  = EcalClusterTools::eMax(seedCluster,recHits)/raw_energy;  //scSeedEmax/scRawEnergy
+    eval[7]  = EcalClusterTools::e2nd(seedCluster,recHits)/raw_energy; // scSeedE2nd/scRawEnergy
+    eval[8] = (eLeft + eRight != 0.f  ? (eLeft-eRight)/(eLeft+eRight) : 0.f); //scSeedLeftRightAsym
+    eval[9] = (eTop  + eBottom != 0.f ? (eTop-eBottom)/(eTop+eBottom) : 0.f); //scSeedTopBottomAsym
+    eval[10] = sigmaIetaIeta; //scSeedSigmaIetaIeta
+    eval[11] = sigmaIetaIphi; //scSeedSigmaIetaIphi
+    eval[12] = sigmaIphiIphi; //scSeedSigmaIphiIphi
+    eval[13] = std::max(0,numberOfClusters-1); //N_ECALClusters
+    eval[14] = clusterMaxDR; //clusterMaxDR
+    eval[15] = clusterMaxDRDPhi; //clusterMaxDRDPhi
+    eval[16] = clusterMaxDRDEta; //clusterMaxDRDEta
+    eval[17] = clusterMaxDRRawEnergy/raw_energy; //clusterMaxDRRawEnergy/scRawEnergy
+    eval[18] = clusterRawEnergy[0]/raw_energy; //clusterRawEnergy[0]/scRawEnergy
+    eval[19] = clusterRawEnergy[1]/raw_energy; //clusterRawEnergy[1]/scRawEnergy
+    eval[20] = clusterRawEnergy[2]/raw_energy; //clusterRawEnergy[2]/scRawEnergy
+    eval[21] = clusterDPhiToSeed[0]; //clusterDPhiToSeed[0]
+    eval[22] = clusterDPhiToSeed[1]; //clusterDPhiToSeed[1]
+    eval[23] = clusterDPhiToSeed[2]; //clusterDPhiToSeed[2]
+    eval[24] = clusterDEtaToSeed[0]; //clusterDEtaToSeed[0]
+    eval[25] = clusterDEtaToSeed[1]; //clusterDEtaToSeed[1]
+    eval[26] = clusterDEtaToSeed[2]; //clusterDEtaToSeed[2]
```

Comment:
almost all of the commented out text is not a comment and looks more like an automated repeat of the active code.
Please explain why this is needed.